app/assets/javascripts/app/discussions/discussion.js

```diff
@@ -14,25 +14,19 @@
 
     Discussion
       .find(discussionId)
-      .then(updateDiscussion);
+      .then(setDiscussion);
 
-    function toogleDiscussionState(discussion) {
-      discussion.open = !discussion.open;
-      discussion.update();
-    }
-
-    function updateDiscussion(discussion) {
-      resetController(discussion);
-
-      discussion
-        .refresh()
-        .then(resetController);
-    }
-
-    function resetController(discussion) {
+    function setDiscussion(discussion) {
       ctrl.discussion = discussion;
       ctrl.discussion.markAsRead();
       ctrl.pageReady = true;
+
+      discussion.loadComments();
+    }
+
+    function toogleDiscussionState(discussion) {
```

Comment:
I know you haven't changed this method in this PR, but does it need a discussion passed in as parameter? Can't you use ```this.open``` and ```this.update()```?